annlite/executor.py

```diff
@@ -9,57 +9,47 @@
+    :param include_metadata: If True, return the document metadata in response
+    :param limit: Number of results to get for each query document in search
+    :param ef_construction: The construction time/accuracy trade-off
+    :param ef_search: The query time accuracy/speed trade-off
+    :param max_connection: The maximum number of outgoing connections in the
+        graph (the "M" parameter)
+    :param index_access_paths: Default traversal paths on docs
+            (used for indexing, delete and update), e.g. '@r', '@c', '@r,c'
+    :param search_access_paths: Default traversal paths on docs
+    (used for search), e.g. '@r', '@c', '@r,c'
+    :param columns: A list or dict of column names to index.
     """
 
     def __init__(
         self,
         n_dim: int = 0,
         metric: str = 'cosine',
         limit: int = 10,
-        ef_construction: int = 200,
-        ef_query: int = 50,
-        max_connection: int = 16,
+        data_path: Optional[str] = None,
+        ef_construction: Optional[int] = None,
+        ef_search: Optional[int] = None,
+        max_connection: Optional[int] = None,
         include_metadata: bool = True,
         index_access_paths: str = '@r',
         search_access_paths: str = '@r',
-        columns: Optional[Union[Dict, List]] = None,
-        filterable_attrs: Optional[Dict] = None,
-        serialize_config: Optional[Dict] = None,
+        columns: Optional[Union[List[Tuple[str, str]], Dict[str, str]]] = None,
         dim: int = None,
         *args,
         **kwargs,
     ):
-        """
-        :param n_dim: Dimensionality of vectors to index
-        :param dim: Deprecated, use n_dim instead
-        :param metric: Distance metric type. Can be 'euclidean', 'inner_product', or 'cosine'
-        :param include_metadata: If True, return the document metadata in response
-        :param limit: Number of results to get for each query document in search
-        :param ef_construction: The construction time/accuracy trade-off
-        :param ef_query: The query time accuracy/speed trade-off
-        :param max_connection: The maximum number of outgoing connections in the
-            graph (the "M" parameter)
-        :param index_access_paths: Default traversal paths on docs
-                (used for indexing, delete and update), e.g. '@r', '@c', '@r,c'
-        :param search_access_paths: Default traversal paths on docs
-        (used for search), e.g. '@r', '@c', '@r,c'
-        :param columns: A list or dict of column names to index.
-        :param filterable_attrs: Dict of attributes that can be used for filtering. The key is the attribute name and the
-                value is a string that can be parsed as a valid Python type ['float', 'str', 'int']. It only works if
-                `columns` is None.
-        :param serialize_config: The configurations used for serializing documents, e.g., {'protocol': 'pickle'}
-        """
```

Comment:
This is also parsed by jina hub, let's keep params here

Reply:
This part was attached to class itself, not under the `__init__()` (to satisfy the [docstring style](https://github.com/jina-ai/jina/blob/master/.github/DOCSTRINGS.rst#documenting-classes) ).